### PR TITLE
[Yoong Yi En] iP

### PR DESCRIPTION
**Pull request by yienyoong**

### CODE REVIEW

File: src/main/java/duke/Storage.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new DukeException("Failed to save data.");
+        }
+    }
+
+    /**
+     * Populates the task list with the tasks saved in the task data file upon start of program.
+     *
+     * @param taskList The task list to be populated
```
Comment (swayongshen) [inline src/main/java/duke/Storage.java]: Should there by a period at the end of the param description?

**Review by swayongshen — COMMENTED** [review]
Excellent code, can't find anything to nitpick on! Keep up the good work.

File: src/main/java/duke/Storage.java (hunk 1 above)
Comment (yienyoong) [reply src/main/java/duke/Storage.java]: Yes, there should be. I missed it out. Thank you! :)

File: src/main/java/duke/DukeException.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+public class DukeException extends Exception {
+    private static final String TOP_BORDER = "_____________________________________________________________________";
+    private static final String BTM_BORDER = "---------------------------------------------------------------------";
+    private static final String PADDING = "  ";
+
+    /**
+     * Create a new DukeException to be thrown
+     *
+     * @param msg The error message
+     */
```
Comment (kieron560) [inline src/main/java/duke/DukeException.java]: Hi, I believe there should be punctuation at the end of the description, based on the javadocs

**Review by kieron560 — COMMENTED** [review]
Really nice and clean code! :)

File: src/test/java/duke/command/DeadlineCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        });
+    }
+
+    @Test
+    public void run_wronglyFormattedDeadlineCommand2_exceptionThrown() {
+        DeadlineCommand dlCmd = new DeadlineCommand("deadline testWronglyFormatted DeadlineCmd /at 2020-01-01 1800", new Ui());
+        assertThrows(DukeException.class, () -> {
+            dlCmd.run(new StorageStub(), new TaskList());
+        });
+    }
```
Comment (maikongeh) [inline src/test/java/duke/command/DeadlineCommandTest.java]: Your code is well written and understandable. Your testing is also rigorous and complete. Keep up the good work!

**Pull request closed without merging**